### PR TITLE
fix(LVGL port): Fixed monochromatic screen in LVGL9 + example

### PR DESCRIPTION
**Pull request by espzav**
# ESP-BSP Pull Request checklist

- [x] Version of modified component bumped
- [x] CI passing

# Change description
- Fixed using monochromatic screens with LVGL port and LVGL9
- Support for LV_COLOR_FORMAT_I1 for monochromatic screen

### CODE REVIEW

**Review by tore-espressif — APPROVED** [review]
@espzav Changes LGTM.

Although I don't really understand the monochrome color.
Is this something that was broken by LVGL? Should we migrate to their built-in monochrome color?

File: components/esp_lvgl_port/src/lvgl9/esp_lvgl_port_disp.c [hunk 1]
```diff
@@ -297,6 +297,9 @@ static lv_display_t *lvgl_port_add_disp_priv(const lvgl_port_display_cfg_t *disp
 
     /* Monochrome display settings */
     if (disp_cfg->monochrome) {
+        /* Monochrome can be used only in RGB565 color format */
+        ESP_RETURN_ON_FALSE(display_color_format == LV_COLOR_FORMAT_RGB565, NULL, TAG, "Monochrome can be used only in display color format RGB565!");
```
[the remark this answers is not in the document]
Comment (espzav) [reply components/esp_lvgl_port/src/lvgl9/esp_lvgl_port_disp.c:301]: Yes. I haven't got more time for support this feature (`LV_COLOR_DEPTH == 1`) now. But maybe we would have it: https://gitlab.espressif.cn:6688/espressif/esp-idf/-/merge_requests/33508#note_1730387 I will update LVGL port after that :-)

File: components/esp_lvgl_port/idf_component.yml
Comment (tore-espressif) [inline components/esp_lvgl_port/idf_component.yml:1]: Let's discuss LVGL v9.2 support before merging this :)

File: components/esp_lvgl_port/examples/i2c_oled/main/i2c_oled_example_main.c [hunk 2]
```diff
@@ -123,20 +123,29 @@ void app_main(void)
...
     lv_disp_t *disp = lvgl_port_add_disp(&disp_cfg);
 
-    /* Rotation of the screen */
-    lv_disp_set_rotation(disp, LV_DISP_ROT_NONE);
-
     ESP_LOGI(TAG, "Display LVGL Scroll Text");
     // Lock the mutex due to the LVGL APIs are not thread-safe
     if (lvgl_port_lock(0)) {
+        /* Rotation of the screen */
+        lv_disp_set_rotation(disp, LV_DISPLAY_ROTATION_0);
```
Comment (mutatrum) [inline components/esp_lvgl_port/examples/i2c_oled/main/i2c_oled_example_main.c:147]: This change is also LVGL version dependent.

Comment (espzav) [reply components/esp_lvgl_port/examples/i2c_oled/main/i2c_oled_example_main.c:147]: Yes, it is. But we have included this header for LVGL8: https://github.com/espressif/esp-bsp/blob/master/components/esp_lvgl_port/include/esp_lvgl_port_compatibility.h

**Review by DrStone71 — APPROVED** [review]
hi,

with dependencies:
  lvgl/lvgl: "==9.2.2"
  esp_lcd_sh1107: "==1.1.0"
  esp_lvgl_port: "==2.4.2"
  
  i haved make changes and with SSD1306 Display works fine.

Tanks

**Comment by obaranov** [comment]
I also verified the fix on the ssd1306 (128x64), and graphics  now works fine with lvgl9.

```
dependencies:
  espressif/esp_lvgl_port: 
    git: https://github.com/espressif/esp-bsp.git
    path: components/esp_lvgl_port
    version: fix/lvgl_port_monochromatic_v9
```

**Comment by mutatrum** [comment]
Confirmed the fix is working. For the `i2c_oled` example, when moving to LVGL 9, the following line needs to be removed:

https://github.com/espressif/esp-bsp/blob/240d99728ed899226083feda9e98524c110543aa/components/esp_lvgl_port/examples/i2c_oled/sdkconfig.defaults#L5

This might need to be added to the readme of that example.

**Comment by espzav** [comment]
Hi @tore-espressif, I added support for I1 color type for monochromatic screens and fixed some minor issues. PTAL

**Comment by mutatrum** [comment]
Confirmed, example now works with LVGL 9.2.2 and SSD1306 132x32 display. LGTM. Thanks!

File: components/esp_lvgl_port/examples/i2c_oled/main/i2c_oled_example_main.c [hunk 3]
```diff
@@ -52,21 +52,22 @@ extern void example_lvgl_demo_ui(lv_disp_t *disp);
...
+    ESP_ERROR_CHECK(i2c_new_master_bus(&bus_config, &i2c_bus));
 
     ESP_LOGI(TAG, "Install panel IO");
     esp_lcd_panel_io_handle_t io_handle = NULL;
     esp_lcd_panel_io_i2c_config_t io_config = {
         .dev_addr = EXAMPLE_I2C_HW_ADDR,
+        .scl_speed_hz = EXAMPLE_LCD_PIXEL_CLOCK_HZ,
         .control_phase_bytes = 1,               // According to SSD1306 datasheet
         .lcd_cmd_bits = EXAMPLE_LCD_CMD_BITS,   // According to SSD1306 datasheet
         .lcd_param_bits = EXAMPLE_LCD_CMD_BITS, // According to SSD1306 datasheet
```
Comment (mutatrum) [inline components/esp_lvgl_port/examples/i2c_oled/main/i2c_oled_example_main.c:73]: Should be EXAMPLE_LCD_PARAM_BITS constant.

**Comment by mutatrum** [comment]
I've also tested this on an SSH1107 132x64 screen. This works, however, as this screen is oriented differently, I wanted to rotate the screen. Here I noticed the screen rotation is broken in LVGL 9.

With LVGL 8.4, I can just set `lv_disp_set_rotation(disp, LV_DISPLAY_ROTATION_90);` (or another non-0 value) in the example  code and the screen is rotated. With LVGL 9, the application crashes when using 90 or 270. 180 is fine. I haven't dug into the code yet.

**Pull request merged**